portal-kernel/src/com/liferay/portal/kernel/workflow/DefaultWorkflowTransition.java

```diff
@@ -26,16 +26,21 @@ public String getLabel(Locale locale) {
 			return HtmlUtil.escape(label);
 		}
 
-		label = _labelMap.get(LocaleUtil.getSiteDefault());
+		Language language = LanguageUtil.getLanguage();
+
+		label = language.get(locale, (_name != null) ? _name : "proceed");
 
 		if (label != null) {
 			return HtmlUtil.escape(label);
 		}
 
-		Language language = LanguageUtil.getLanguage();
+		label = _labelMap.get(LocaleUtil.getSiteDefault());
 
-		return HtmlUtil.escape(
-			language.get(locale, (_name != null) ? _name : "proceed"));
+		if (label != null) {
+			return HtmlUtil.escape(label);
+		}
+
+		return _name;
```

Comment:
@p-albuquerque this one is a little bit different, I believe it should be this way:

```java
public String getLabel(Locale locale) {
		String label = _labelMap.get(locale);

		if (label != null) {
			return HtmlUtil.escape(label);
		}

		Language language = LanguageUtil.getLanguage();

		if (_name == null) {
			return language.get(locale, "proceed");
		}

		label = language.get(locale, _name, null);

		if (label != null) {
			return HtmlUtil.escape(label);
		}

		label = _labelMap.get(LocaleUtil.getSiteDefault());

		if (label != null) {
			return HtmlUtil.escape(label);
		}

		return _name;
	}
```

Reply:
Got it